sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java

```diff
@@ -642,19 +642,39 @@ private static String getHttpUrlFromServerSpanStableSemconv(Attributes attribute
         if (scheme == null) {
             return null;
         }
-        String path = attributes.get(SemanticAttributes.URL_PATH);
-        if (path == null) {
-            return null;
-        }
         String host = attributes.get(SemanticAttributes.SERVER_ADDRESS);
         if (host == null) {
             return null;
         }
         Long port = attributes.get(SemanticAttributes.SERVER_PORT);
+        String path = attributes.get(SemanticAttributes.URL_PATH);
+        if (path == null) {
+            return null;
+        }
+        String query = attributes.get(SemanticAttributes.URL_QUERY);
+
+        int len = scheme.length() + host.length() + path.length();
+        if (port != null) {
+            len += 6; // max 5 digits for port plus the port separator ":"
+        }
+        if (query != null) {
+            len += query.length() + 1; // including the query separator "?"
+        }
```

Comment:
Extract  method from line 656 to line 662 with a `computeFullpathLength(...)` method name?

Reply:
I sort of like to be able to see the corresponding ifs here and below when appending